---
title: "Single Cell on AWS (ScAWS)"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{ScAWS_Vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

#control vs disease condition: paper ppt slide
#Bullet list of whow and what using
#uml of what's happening with scaws image - fastqc figures
#scaws vignette <it's what we're reading>
#figures/shiny/findings
#reproducible images, shippable/easy hosting package and apps, github, cost

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

In this vignette, we have set up the instance, run our setup and generated count matrices. We now use 

```{r setup}
#sessionInfo()
library(ScAWS)
#Set color palette
grey="grey"
pal=ggpubr::get_palette("npg", k=40)
grey_pal=c("grey", pal[1])
bi=pal[c(2,5)]

#Which reference to use, Monaco is a PBMC dataset
ref=celldex::MonacoImmuneData()
set.seed(1)
#Detect number of cores and use future to use parallelized seurat functions
future::plan("multiprocess", workers = parallel::detectCores())

#Increase globals size as Seurat exceeds max size in common use cases
options(future.globals.maxSize = 4000 * 1024^2)
```

```{r}
setwd("~/Desktop/data/")
# Lists sample directories generated with STAR
dirs=list_dirs("~/Desktop/data/")
npcs=30
dims=30
# Filtered or raw UMI counts are located in the STAR directory {sample}/Solo.out/Gene/filtered or {sample}/Solo.out/Gene/raw, respectively.
star_matrix_list=make_star_matrix_list(dirs, matrix_type = "filtered")

seurat_object_list=make_seurat_object_list(star_matrix_list, singler_ref=ref, singler_labels=ref$label.main)

seurat_object_list=seurat_process_list(seurat_object_list, npcs=npcs, dims=dims)
integrated_seurat_object=integrate_seurat_object_list(seurat_object_list, npcs=npcs, dims=dims)
```

```{r}
p1=Seurat::DimPlot(integrated_seurat_object, reduction = "umap"
           #, group.by = "stim"
           )
#p2=DimPlot(integrated_seurat_object, reduction = "umap", label = TRUE, repel = TRUE)
#p1 + p2
p1
```


